e2e/cypress/integration/multi_team_and_dm/multi_team_spec.js

```diff
@@ -0,0 +1,143 @@
+                userC = otherUser;
+                cy.apiAddUserToTeam(team.id, userC.id);
+            });
+
+            cy.apiCreateTeam('team', 'Team').then(({team: otherTeam}) => {
+                teamB = otherTeam;
+                cy.apiAddUserToTeam(otherTeam.id, userA.id);
+                cy.apiAddUserToTeam(otherTeam.id, userB.id);
+            });
+        });
+    });
+
+    beforeEach(() => {
+        // # Log in to Team A with an account that has joined multiple teams.
+        cy.apiLogin(userA);
+
+        // # On an account on two teams, view Team A
+        cy.visit(`/${teamA.name}/channels/town-square`);
+    });
+
+    it('MM-T433 Switch teams', () => {
+        // # Open several DM channels, including accounts that are not on Team B.
+        cy.apiCreateDirectChannel([userA.id, userB.id]).then(() => {
+            cy.visit(`/${teamA.name}/channels/${userA.id}__${userB.id}`).wait(TIMEOUTS.FIVE_SEC);
+            cy.postMessage(':)');
+        });
+        cy.apiCreateDirectChannel([userA.id, userC.id]).then(() => {
+            cy.visit(`/${teamA.name}/channels/${userA.id}__${userC.id}`).wait(TIMEOUTS.FIVE_SEC);
+            cy.postMessage(':(');
+        });
+
+        // # Click Team B in the team sidebar.
+        cy.get(`#${teamB.name}TeamButton`, {timeout: TIMEOUTS.ONE_MIN}).should('be.visible').click();
+
+        // * Channel list in the LHS is scrolled to the top.
+        cy.get('#publicChannelList').get('.active').should('contain', 'Town Square');
+
+        // * Verify team display name changes correctly.
+        cy.get('#headerTeamName', {timeout: TIMEOUTS.ONE_MIN}).should('contain', teamB.display_name);
+
+        // * DM Channel list should be the same on both teams with no missing names.
+        cy.get('#directChannelList').findByText(`${userB.username}`).should('be.visible');
+        cy.get('#directChannelList').findByText(`${userC.username}`).should('be.visible');
+
+        // # Post a message in Town Square in Team B
+        cy.postMessage('Hello World');
+
+        // * Verify posting a message works properly.
+        cy.getLastPostId().then((postId) => {
+            cy.get(`#postMessageText_${postId}`).should('be.visible').and('have.text', 'Hello World');
+        });
+
+        // # Click Team A in the team sidebar.
+        cy.get(`#${teamA.name}TeamButton`, {timeout: TIMEOUTS.ONE_MIN}).should('be.visible').click();
+
+        // * Verify there is no cross contamination between teams.
+        cy.get('#sidebarItem_town-square').should('not.have.class', 'unread-title');
+
+        // * DM Channel list should be the same on both teams with no missing names.
+        cy.get('#directChannelList').findByText(`${userB.username}`).should('be.visible');
+        cy.get('#directChannelList').findByText(`${userC.username}`).should('be.visible');
+
+        // * Channel viewed on team a before switching should be the one that displays after switching back (Town Square does not briefly show).
```

Comment:
on _a_ team.

Also - what does "Town square does not briefly show" mean?

Reply:
I believe it means that it should not open Town Square after we return back to Team A.

Reply:
Oh it's a statement. I thought it was a comment about some quirk.